backend-shared/config/db/util/utils_test.go

```diff
@@ -1,12 +1,881 @@
+			Expect(err).To(BeNil())
+
+			workspace := v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-user",
+					UID:       "",
+					Namespace: "test-namespace",
+				},
+				Spec: v1.NamespaceSpec{},
+			}
+
+			managedEnvironment, isNew, err := GetOrCreateManagedEnvironmentByNamespaceUID(ctx, workspace, dbQueries, log)
+
+			Expect(err).NotTo(BeNil())
+			Expect(isNew).To(BeFalse())
+			Expect(managedEnvironment).To(BeNil())
+		})
+	})
+
+	Context("Testing for GetOrCreateDeploymentToApplicationMapping function.", func() {
+
+		var err error
+		var log logr.Logger
+		var ctx context.Context
+		var application db.Application
+		var dbQueries db.AllDatabaseQueries
+		var clusterCredentials db.ClusterCredentials
+		var managedEnvironment db.ManagedEnvironment
+		var gitopsEngineCluster db.GitopsEngineCluster
+		var gitopsEngineInstance db.GitopsEngineInstance
+		var deploymentToApplicationMapping *db.DeploymentToApplicationMapping
+
+		BeforeEach(func() {
+			ctx, dbQueries, log, _, err = initialSetUp()
+			Expect(err).To(BeNil())
+
+			// ----------------------------------------------------------------------------
+			By("First create resources required by DeploymentToApplicationMapping resource.")
+			// ----------------------------------------------------------------------------
+
+			// Create ClusterCredentials
+			clusterCredentials = db.ClusterCredentials{
+				Host:                        "host",
+				Kube_config:                 "kube-config",
+				Kube_config_context:         "kube-config-context",
+				Serviceaccount_bearer_token: "serviceaccount_bearer_token",
+				Serviceaccount_ns:           "Serviceaccount_ns",
+			}
+			err = dbQueries.CreateClusterCredentials(ctx, &clusterCredentials)
+			Expect(err).To(BeNil())
+
+			// ---------------------------------------------------------------------------
+			// Create ManagedEnvironment
+
+			managedEnvironment = db.ManagedEnvironment{
+				Clustercredentials_id: clusterCredentials.Clustercredentials_cred_id,
+				Name:                  "my-managed-environment",
+			}
+			err = dbQueries.CreateManagedEnvironment(ctx, &managedEnvironment)
+			Expect(err).To(BeNil())
+
+			// ---------------------------------------------------------------------------
+			// Create GitopsEngineCluster
+
+			gitopsEngineCluster = db.GitopsEngineCluster{
+				Clustercredentials_id: clusterCredentials.Clustercredentials_cred_id,
+			}
+			err = dbQueries.CreateGitopsEngineCluster(ctx, &gitopsEngineCluster)
+			Expect(err).To(BeNil())
+
+			// ---------------------------------------------------------------------------
+			// Create GitopsEngineInstance
+
+			gitopsEngineInstance = db.GitopsEngineInstance{
+				Namespace_name:   "my-namespace",
+				Namespace_uid:    "test-1",
+				EngineCluster_id: gitopsEngineCluster.Gitopsenginecluster_id,
+			}
+			err = dbQueries.CreateGitopsEngineInstance(ctx, &gitopsEngineInstance)
+			Expect(err).To(BeNil())
+
```

Comment:
```suggestion
clusterCredentials, managedEnvironment, gitopsEngineCluster, gitopsEngineInstance, _, err := db.CreateSampleData(dbq)
Expect(err).To(BeNil())

```
Rest we can leave as it is

Reply:
Yes I tried that too, but I get error but I get error `primary key should be empty` from the function, so I didnt want to change anything in function so I am created required resources, it also allowed me to give resource details (namespace, name etc) as I need.
Could you please try it once, to confirm that I am not doing something wrong.

Reply:
Sorry just saw your reply on this, will give this a try to reciprocate why it is happening. Github should add system notifications when the tabs are open :(